### PR TITLE
improve logging for modules

### PR DESCRIPTION
**Pull request by tobes**
fix for #1028

make `py3.log()` more like `log()`

now we just do the `pformat` on tuple, list, dict, set

### CODE REVIEW

**Comment by lasers** [comment]
Should we have `self.py3.pl()` where it just prints a line? 
Should we have `self.py3.pl('block')` where it just prints a line `────────── block ─────` ?

I print/log a lot when making new modules. 😞

**Comment by tobes** [comment]
> Should we have self.py3.pl() where it just prints a line?

I'm not convinced there is a real case for this.  It depends what you are doing/debugging module.

The logging says which module is being logged which is the most important thing!

not quite the same but isn't this enough
```
self.py3.log('-' * 40)
self.py3.log('block')
```

**Comment by lasers** [comment]
Ha... Yeah, that'll work. Clever. Much appreciated, @tobes. I might just stick with `('----------...`.

@ultrabug Can we merge this and https://github.com/ultrabug/py3status/pull/1033 today? Many thanks!

**Comment by tobes** [comment]
Could give a concrete example of why this is a problem?

Your solution also only works for lists and if this is a real issue then maybe we should fix it properly.

personally I struggle to see what this helps with, but maybe if I understood why you see it as a problem that would help.

**Comment by lasers** [comment]
### NOW ###
```
:~/src/py3status/py3status/modules] newmodule-secret(+18/-7,+11/-19)+ ± python3 secret.py
Module `test_module`: BEFORE BUFFERS
-------------------
Module `test_module`:
1d [chris:~/src/py3status/py3status/modules] newmodule-secret(+21/-8,+11/-19)+ ± python3 secret.py\012Module `test_module`: BEFORE BUFFERS\012-------------------\012Module `test_module`:\0121d [chris:~/s...837uffer0022
1d [chris:~/src/py3status/py3status/modules] newmodule-secret(+21/-8,+11/-19)+ ± python3 secret.py\012Module `test_module`: BEFORE BUFFERS\012-------------------\012Module `test_module`:\012buffer0018540...351uffer0021
\012^C%                                                                                                                                                                                                 ...54uffer0018
x1b\\x1ebuffer0008',\012 '\\x1f\\x1bt\\x1b\\x1e1\\x1b\\x1ebuffer0007',\012 '\\x1f\\x1bHi\\\\\\\\n\\\\\\\\n\\\\\\\\n\\\\012\\x1b\\x1e9\\x1b\\x1ebuffer0006',\012 '\\x1f\\x1b128     Run module in test mo...71uffer0017
1d [chris:~/src/py3status/py3status/modules] newmodule-secret(+17/-7,+11/-19)+ ± python3 secret.py\012Module `test_module`: BEFORE BUFFERS\012-------------------\012Module `test_module`:\01275uffer0016
'\\x1f\\x1bget back to the original bra\\x1b\\x1ebuffer0008\\x1b\\x1e28',\0128uffer0015
'\\x1f\\x1b \\x1b\\x1ebuffer0009\\x1b\\x1e1',9uffer0014
Compressing objects: 100% (5/5), done.\012Writing objects: 100% (5/5), 576 bytes | 576.00 KiB/s, done.\012Total 5 (delta 4), reused 0 (delta 0)\012remote: Resolving deltas: 100% (4/4), completed with ...39uffer0013
o https://github.com/lasers/py3status.git\012 + eb8977e6...f32b9ffc group-add-placeholders -> group-add-placeholders (forced update)\01230uffer0012
              if self.symmetric is True:\012                  split = int(self.max_size / 2) - 1\0122d [chris:~/src/py3status/py3status/modules] b63f26ec(+8/-2)+ ±\0122d [chris:~/src/py3status/py3sta...24uffer0011
sk space                           [#################################] 100%\012:: Processing package changes...\012(1/1) installing screenkey                                    [######################...71uffer0010
 uffer0009
get back to the original bra8uffer0008
tuffer0007
Hi\\n\\n\\n\012uffer0006
128     Run module in test mode.\0123uffer0004
1d [chris:~/src/py3status/py3status/modules] b63f26ec+* ±\0129uffer0003

-------------------
Module `test_module`: AFTER BUFFERS
Module `test_module`:
['1d [chris:~/src/py3status/py3status/modules] newmodule-secret(+21/-8,+11/-19)+ '
 '± python3 secret.py\\012Module `test_module`: BEFORE '
 'BUFFERS\\012-------------------\\012Module `test_module`:\\0121d '
 '[chris:~/s...\x1b\x1e'
 '4837\x1b\x1e'
 'buffer0022',
 '\x1f\x1b1d [chris:~/src/py3status/py3status/modules] '
 'newmodule-secret(+21/-8,+11/-19)+ ± python3 secret.py\\012Module `test_module`: '
 'BEFORE BUFFERS\\012-------------------\\012Module '
 '`test_module`:\\012buffer0018540...\x1b\x1e'
 '4351\x1b\x1e'
 'buffer0021',
 '\x1f'
 '\x1b\\012^C%                                                                                                                                                                                                 '
 '...\x1b\x1e'
 '354\x1b\x1e'
 'buffer0018',
 "\x1f\x1bx1b\\\\x1ebuffer0008',\\012 "
 "'\\\\x1f\\\\x1bt\\\\x1b\\\\x1e1\\\\x1b\\\\x1ebuffer0007',\\012 "
 "'\\\\x1f\\\\x1bHi\\\\\\\\\\\\\\\\n\\\\\\\\\\\\\\\\n\\\\\\\\\\\\\\\\n\\\\\\\\012\\\\x1b\\\\x1e9\\\\x1b\\\\x1ebuffer0006',\\012 "
 "'\\\\x1f\\\\x1b128     Run module in test mo...\x1b\x1e"
 '171\x1b\x1e'
 'buffer0017',
 '\x1f\x1b1d [chris:~/src/py3status/py3status/modules] '
 'newmodule-secret(+17/-7,+11/-19)+ ± python3 secret.py\\012Module `test_module`: '
 'BEFORE BUFFERS\\012-------------------\\012Module `test_module`:\\012\x1b\x1e'
 '175\x1b\x1e'
 'buffer0016',
 "\x1f\x1b'\\\\x1f\\\\x1bget back to the original "
 "bra\\\\x1b\\\\x1ebuffer0008\\\\x1b\\\\x1e28',\\012\x1b\x1e"
 '68\x1b\x1e'
 'buffer0015',
 "\x1f\x1b'\\\\x1f\\\\x1b \\\\x1b\\\\x1ebuffer0009\\\\x1b\\\\x1e1',\x1b\x1e"
 '39\x1b\x1e'
 'buffer0014',
 '\x1f\x1bCompressing objects: 100% (5/5), done.\\012Writing objects: 100% '
 '(5/5), 576 bytes | 576.00 KiB/s, done.\\012Total 5 (delta 4), reused 0 '
 '(delta 0)\\012remote: Resolving deltas: 100% (4/4), completed with '
 '...\x1b\x1e'
 '339\x1b\x1e'
 'buffer0013',
 '\x1f\x1bo https://github.com/lasers/py3status.git\\012 + eb8977e6...f32b9ffc '
 'group-add-placeholders -> group-add-placeholders (forced update)\\012\x1b\x1e'
 '130\x1b\x1e'
 'buffer0012',
 '\x1f\x1b              if self.symmetric is True:\\012                  split '
 '= int(self.max_size / 2) - 1\\0122d '
 '[chris:~/src/py3status/py3status/modules] b63f26ec(+8/-2)+ ±\\0122d '
 '[chris:~/src/py3status/py3sta...\x1b\x1e'
 '224\x1b\x1e'
 'buffer0011',
 '\x1f\x1bsk space                           '
 '[#################################] 100%\\012:: Processing package '
 'changes...\\012(1/1) installing screenkey                                    '
 '[######################...\x1b\x1e'
 '371\x1b\x1e'
 'buffer0010',
 '\x1f\x1b \x1b\x1e1\x1b\x1ebuffer0009',
 '\x1f\x1bget back to the original bra\x1b\x1e28\x1b\x1ebuffer0008',
 '\x1f\x1bt\x1b\x1e1\x1b\x1ebuffer0007',
 '\x1f\x1bHi\\\\n\\\\n\\\\n\\012\x1b\x1e9\x1b\x1ebuffer0006',
 '\x1f\x1b128     Run module in test mode.\\012\x1b\x1e33\x1b\x1ebuffer0004',
 '\x1f\x1b1d [chris:~/src/py3status/py3status/modules] b63f26ec+* '
 '±\\012\x1b\x1e'
 '59\x1b\x1e'
 'buffer0003',
 '\x1f\x1b']
-------------------
END
{'full_text': '[secret]', 'color': '#00FF00', 'cached_until': 1504525936.0}
^C%                                                                                                                                                                                                                                                                              1d [chris:~/src/py3status/py3status/modules] newmodule-secret(+18/-7,+11/-19)+ ±
```

### DIFF ###
```
1d [chris:~/src/py3status/py3status/modules] newmodule-secret(+21/-8,+11/-19)+ ± python3 secret.py
Module `test_module`: BEFORE BUFFERS
-------------------
Module `test_module`:
1d [chris:~/src/py3status/py3status/modules] newmodule-secret(+21/-8,+11/-19)+ ± python3 secret.py\012Module `test_module`: BEFORE BUFFERS\012-------------------\012Module `test_module`:\0121d [chris:~/s...uffer0022837
1d [chris:~/src/py3status/py3status/modules] newmodule-secret(+21/-8,+11/-19)+ ± python3 secret.py\012Module `test_module`: BEFORE BUFFERS\012-------------------\012Module `test_module`:\012buffer0018540...uffer0021351
\012^C%                                                                                                                                                                                                 ...uffer001854
x1b\\x1ebuffer0008',\012 '\\x1f\\x1bt\\x1b\\x1e1\\x1b\\x1ebuffer0007',\012 '\\x1f\\x1bHi\\\\\\\\n\\\\\\\\n\\\\\\\\n\\\\012\\x1b\\x1e9\\x1b\\x1ebuffer0006',\012 '\\x1f\\x1b128     Run module in test mo...uffer001771
1d [chris:~/src/py3status/py3status/modules] newmodule-secret(+17/-7,+11/-19)+ ± python3 secret.py\012Module `test_module`: BEFORE BUFFERS\012-------------------\012Module `test_module`:\012uffer001675
'\\x1f\\x1bget back to the original bra\\x1b\\x1ebuffer0008\\x1b\\x1e28',\012uffer00158
'\\x1f\\x1b \\x1b\\x1ebuffer0009\\x1b\\x1e1',uffer00149
Compressing objects: 100% (5/5), done.\012Writing objects: 100% (5/5), 576 bytes | 576.00 KiB/s, done.\012Total 5 (delta 4), reused 0 (delta 0)\012remote: Resolving deltas: 100% (4/4), completed with ...uffer001339
o https://github.com/lasers/py3status.git\012 + eb8977e6...f32b9ffc group-add-placeholders -> group-add-placeholders (forced update)\012uffer001230
              if self.symmetric is True:\012                  split = int(self.max_size / 2) - 1\0122d [chris:~/src/py3status/py3status/modules] b63f26ec(+8/-2)+ ±\0122d [chris:~/src/py3status/py3sta...uffer001124
sk space                           [#################################] 100%\012:: Processing package changes...\012(1/1) installing screenkey                                    [######################...uffer001071
 uffer0009
get back to the original brauffer00088
tuffer0007
Hi\\n\\n\\n\012uffer0006
128     Run module in test mode.\012uffer00043
1d [chris:~/src/py3status/py3status/modules] b63f26ec+* ±\012uffer00039

-------------------
Module `test_module`: AFTER BUFFERS
Module `test_module`: ['1d [chris:~/src/py3status/py3status/modules] newmodule-secret(+21/-8,+11/-19)+ ± python3 secret.py\\012Module `test_module`: BEFORE BUFFERS\\012-------------------\\012Module `test_module`:\\0121d [chris:~/s...\x1b\x1ebuffer0022\x1b\x1e4837', '\x1f\x1b1d [chris:~/src/py3status/py3status/modules] newmodule-secret(+21/-8,+11/-19)+ ± python3 secret.py\\012Module `test_module`: BEFORE BUFFERS\\012-------------------\\012Module `test_module`:\\012buffer0018540...\x1b\x1ebuffer0021\x1b\x1e4351', '\x1f\x1b\\012^C%                                                                                                                                                                                                 ...\x1b\x1ebuffer0018\x1b\x1e354', "\x1f\x1bx1b\\\\x1ebuffer0008',\\012 '\\\\x1f\\\\x1bt\\\\x1b\\\\x1e1\\\\x1b\\\\x1ebuffer0007',\\012 '\\\\x1f\\\\x1bHi\\\\\\\\\\\\\\\\n\\\\\\\\\\\\\\\\n\\\\\\\\\\\\\\\\n\\\\\\\\012\\\\x1b\\\\x1e9\\\\x1b\\\\x1ebuffer0006',\\012 '\\\\x1f\\\\x1b128     Run module in test mo...\x1b\x1ebuffer0017\x1b\x1e171", '\x1f\x1b1d [chris:~/src/py3status/py3status/modules] newmodule-secret(+17/-7,+11/-19)+ ± python3 secret.py\\012Module `test_module`: BEFORE BUFFERS\\012-------------------\\012Module `test_module`:\\012\x1b\x1ebuffer0016\x1b\x1e175', "\x1f\x1b'\\\\x1f\\\\x1bget back to the original bra\\\\x1b\\\\x1ebuffer0008\\\\x1b\\\\x1e28',\\012\x1b\x1ebuffer0015\x1b\x1e68", "\x1f\x1b'\\\\x1f\\\\x1b \\\\x1b\\\\x1ebuffer0009\\\\x1b\\\\x1e1',\x1b\x1ebuffer0014\x1b\x1e39", '\x1f\x1bCompressing objects: 100% (5/5), done.\\012Writing objects: 100% (5/5), 576 bytes | 576.00 KiB/s, done.\\012Total 5 (delta 4), reused 0 (delta 0)\\012remote: Resolving deltas: 100% (4/4), completed with ...\x1b\x1ebuffer0013\x1b\x1e339', '\x1f\x1bo https://github.com/lasers/py3status.git\\012 + eb8977e6...f32b9ffc group-add-placeholders -> group-add-placeholders (forced update)\\012\x1b\x1ebuffer0012\x1b\x1e130', '\x1f\x1b              if self.symmetric is True:\\012                  split = int(self.max_size / 2) - 1\\0122d [chris:~/src/py3status/py3status/modules] b63f26ec(+8/-2)+ ±\\0122d [chris:~/src/py3status/py3sta...\x1b\x1ebuffer0011\x1b\x1e224', '\x1f\x1bsk space                           [#################################] 100%\\012:: Processing package changes...\\012(1/1) installing screenkey                                    [######################...\x1b\x1ebuffer0010\x1b\x1e371', '\x1f\x1b \x1b\x1ebuffer0009\x1b\x1e1', '\x1f\x1bget back to the original bra\x1b\x1ebuffer0008\x1b\x1e28', '\x1f\x1bt\x1b\x1ebuffer0007\x1b\x1e1', '\x1f\x1bHi\\\\n\\\\n\\\\n\\012\x1b\x1ebuffer0006\x1b\x1e9', '\x1f\x1b128     Run module in test mode.\\012\x1b\x1ebuffer0004\x1b\x1e33', '\x1f\x1b1d [chris:~/src/py3status/py3status/modules] b63f26ec+* ±\\012\x1b\x1ebuffer0003\x1b\x1e59', '\x1f\x1b']
-------------------
END
{'full_text': '[secret]', 'color': '#00FF00', 'cached_until': 1504525875.0}
^C%
```

The problem is when I'm trying to split up an output into a list-line first... then splitting the list-line up into list-fields... It's hard / useless to parse all this formatted lists with list-fields being broken up into more things....

If the lists are over 80, then we assume that it's not ready to be formatted... so I just want to make sure I'm splitting things up correctly... without having it splitting up anything prematurely. This might be easier if you copy all the log here into vim... The diff doesn't split anything up... so I can just know it's currently in the list right now.

**Comment by tobes** [comment]
When I say an example I mean a minimal one.  what is useful is the exact thing you are trying to log by that I mean what `repr(thing)` will give.  Seeing the output does not help because I've no idea what you are actually outputting.

Incidentally your patch has a python2 bug.

I still do not understand what you really want I suspect `self.py3.log(repr(thing))` is all you need that way there should be no reformatting as you have given a str.  It feels like you are trying to hard.

I'm also not sure why you are logging stuff if you do not want to see it.

**Comment by lasers** [comment]
`self.py3.log(repr(thing))` will work okay here. Without going into details, I'm trying to work out the separators and making sure it's still there... as I find that `str.splitlines()` can eat them. Having everything broken up into pieces makes things hard to parse.... especially when most of them are just long lines and/or long lists.

>When I say an example I mean a minimal one

This is the output I'm currently working with right now. Not wanting to change. Sorry. .__.

>Incidentally your patch has a python2 bug.

No problem. I don't use python2. I'll use `self.py3.log(repr(thing))` for the list. 👍 

>I'm also not sure why you are logging stuff if you do not want to see it.

Debugging. I will remove them later when everything fits in just perfect. Cheers.
Also, less works for `py3` not having to come up with a `max` in the list. This is fine now.

**Comment by lasers** [comment]
New issue / question / discussion.
```
TypeError: argument of type 'float' is not iterable
  File "/home/chris/src/py3status/py3status/module.py", line 729, in run
    response = method()
  File "/home/chris/src/py3status/py3status/py3.py", line 449, in log
    if '\n' in message:
```
This occurs when we `self.py3.log(int(number))` or `self.py3.log(float(number))` because of that `if '\n' in message:` string. I don't think that happened before... but then again, I don't always log the numbers. Anyhow, just a heads up.

**Comment by tobes** [comment]
@lasers thanks should be fixed now
